bots/pr/src/test/java/org/openjdk/skara/bots/pr/ReviewersTests.java

```diff
@@ -613,4 +613,73 @@ void testZeroReviewer(TestInfo testInfo) throws IOException {
             assertTrue(reviewerPr.body().contains(getReviewersExpectedProgress(0, 0, 0, 0, 0)));
         }
     }
+
+    @Test
+    void testReviewCommentsAfterApprovedReview(TestInfo testInfo) throws IOException {
+        try (var credentials = new HostCredentials(testInfo);
+             var tempFolder = new TemporaryDirectory()) {
+            var author = credentials.getHostedRepository();
+            var integrator = credentials.getHostedRepository();
+            var bot = credentials.getHostedRepository();
+
+            var censusBuilder = credentials.getCensusBuilder()
+                    .addReviewer(integrator.forge().currentUser().id())
+                    .addAuthor(author.forge().currentUser().id());
+            var prBot = PullRequestBot.newBuilder().repo(bot).censusRepo(censusBuilder.build()).build();
+
+            // Populate the projects repository
+            var localRepoFolder = tempFolder.path().resolve("localrepo");
+            var localRepo = CheckableRepository.init(localRepoFolder, author.repositoryType());
+            var masterHash = localRepo.resolve("master").orElseThrow();
+            assertFalse(CheckableRepository.hasBeenEdited(localRepo));
+            localRepo.push(masterHash, author.url(), "master", true);
+
+            // Make a change with a corresponding PR
+            var editHash = CheckableRepository.appendAndCommit(localRepo);
+            localRepo.push(editHash, author.url(), "edit", true);
+            var pr = credentials.createPullRequest(author, "master", "edit", "123: This is a pull request");
+
+            var reviewerPr = integrator.pullRequest(pr.id());
+
+            // Approve it as another user
+            reviewerPr.addReview(Review.Verdict.APPROVED, "Approved");
+            TestBotRunner.runPeriodicItems(prBot);
+            TestBotRunner.runPeriodicItems(prBot);
+            // The pr should contain 'Ready' label
+            var updatedPr = author.pullRequest(pr.id());
+            assertTrue(updatedPr.labelNames().contains("ready"));
+
+            // Add a review comment
+            reviewerPr.addReview(Review.Verdict.NONE, "Just a comment1");
+            TestBotRunner.runPeriodicItems(prBot);
+            TestBotRunner.runPeriodicItems(prBot);
+            // The pr should still contain 'Ready' label
+            updatedPr = author.pullRequest(pr.id());
+            assertTrue(updatedPr.labelNames().contains("ready"));
```

Comment:
I am still wondering why we should update the pr manually.

Reply:
An instance of `PullRequest` is not guaranteed to reflect the current server state of the pr, it should always be considered a snapshot. This property of a `PullRequest` wasn't properly reflected in the `TestPullRequest` implementation until recently, with the explicit `TestPullRequestStore` object that represents the server side state.

In test code, instead of explicitly fetching an updated snapshot of a PR, you can access the store directly to inspect the state through `pr.store()`. Most old test code has not been updated to use this. Note that the `TestPullRequestStore` doesn't have as many convenience methods for all the data as `TestPullRequest` has, so it may be a bit clunky, at least for labels. You could always add a better accessor (in `TestIssueStore` in this case) to help with that.

Reply:
> An instance of `PullRequest` is not guaranteed to reflect the current server state of the pr, it should always be considered a snapshot. This property of a `PullRequest` wasn't properly reflected in the `TestPullRequest` implementation until recently, with the explicit `TestPullRequestStore` object that represents the server side state.
> 
> In test code, instead of explicitly fetching an updated snapshot of a PR, you can access the store directly to inspect the state through `pr.store()`. Most old test code has not been updated to use this. Note that the `TestPullRequestStore` doesn't have as many convenience methods for all the data as `TestPullRequest` has, so it may be a bit clunky, at least for labels. You could always add a better accessor (in `TestIssueStore` in this case) to help with that.

Got it! Thanks for explaining this to me. I will use `pr.store()` to do the test.